src/main/java/graphql/execution/ConditionalNodes.java

```diff
@@ -28,19 +25,23 @@ public boolean shouldInclude(Map<String, Object> variables, List<Directive> dire
     }
 
     private boolean getDirectiveResult(Map<String, Object> variables, List<Directive> directives, String directiveName, boolean defaultValue) {
-        List<Directive> foundDirectives = getDirectiveByName(directives, directiveName);
-        if (!foundDirectives.isEmpty()) {
-            Directive directive = foundDirectives.get(0);
-            Map<String, Object> argumentValues = valuesResolver.getArgumentValues(SkipDirective.getArguments(), directive.getArguments(), variables);
+        Directive foundDirective = findDirectiveByName(directives, directiveName);
+        if (foundDirective != null) {
+            Map<String, Object> argumentValues = valuesResolver.getArgumentValues(SkipDirective.getArguments(), foundDirective.getArguments(), variables);
             Object flag = argumentValues.get("if");
             Assert.assertTrue(flag instanceof Boolean, () -> String.format("The '%s' directive MUST have a value for the 'if' argument", directiveName));
             return (Boolean) flag;
         }
         return defaultValue;
     }
 
-    private List<Directive> getDirectiveByName(List<Directive> directives, String name) {
-        return NodeUtil.allDirectivesByName(directives).getOrDefault(name, emptyList());
+    private Directive findDirectiveByName(List<Directive> directives, String name) {
```

Comment:
Move the logic of `findDirectiveByName` to `NodeUtil`, and replace all `getXXByName` with `getNodeByName(List<NamedNode> list, String name)` will be better.

Reply:
done